File: module_test/src/main/java/com/gas/test/ui/test/ArticlePresenter.java
package com.gas.test.ui.test;

import android.app.Application;

import com.base.lib.di.scope.FragmentScope;
import com.base.lib.mvp.BasePresenter;
import com.base.lib.mvp.IModel;

import javax.inject.Inject;

@FragmentScope
public class ArticlePresenter extends BasePresenter<IModel,ArticleContract.View> {


    @Inject
    Application mApplication;


    @Inject
    public ArticlePresenter( ArticleContract.View rootView) {
        super( rootView);
    }


    @Override
    public void onDestroy() {
        mApplication = null;
        super.onDestroy();
    }

}

File: module_beauty/src/main/java/com/gas/beauty/ui/article/mvp/ArticlePresenter.kt
package com.gas.beauty.ui.article.mvp;

import android.app.Application;

import com.base.lib.di.scope.FragmentScope;
import com.base.lib.mvp.BasePresenter;

import javax.inject.Inject;

@FragmentScope
public class ArticlePresenter extends BasePresenter<ArtileModle, ArticleContract.View> {


    @Inject
    Application mApplication;


    @Inject
    public ArticlePresenter(ArtileModle model, ArticleContract.View rootView) {
        super(model, rootView);
    }


    @Override
    public void onDestroy() {
        mApplication = null;
        super.onDestroy();
    }

}
